### PR TITLE
Board nucleo f103

### PR DESCRIPTION
**Pull request by lebrush**
Initial support for the board ST Nucleo f103

### CODE REVIEW

File: boards/nucleo-f103/Makefile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+MODULE =$(BOARD)_base
```
Comment (thomaseichinger) [inline boards/nucleo-f103/Makefile]: Could you please change this to `MODULE = board` in reference to upcoming #4376?

File: boards/nucleo-f103/include/board.h [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    boards_nucleo-f103 Nucleo-F103
+ * @ingroup     boards
+ * @brief       Board specific files for the nucleo-f303 board
```
Comment (thomaseichinger) [inline boards/nucleo-f103/include/board.h]: s/nucleo-f303/nucleo-f103/

File: boards/nucleo-f103/include/board.h [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+ * @{
+ *
+ * @file
+ * @brief       Board specific definitions for the nucleo-f303 board
```
Comment (thomaseichinger) [inline boards/nucleo-f103/include/board.h]: As above.

**Comment by thomaseichinger** [comment]
@lebrush Great work. Could you please address the minor comments above?

**Comment by lebrush** [comment]
@thomaseichinger thanks for the quick review! All your comments are implemented now.

**Comment by lebrush** [comment]
@thomaseichinger seems tracis was cancelled... can you restart it? thanks!

**Comment by OlegHahm** [comment]
Travis failed because there are some applications that are too big for this board and because there are some warnings. Please check https://travis-ci.org/RIOT-OS/RIOT/jobs/95527289 for details.

**Comment by lebrush** [comment]
@OlegHahm thanks for starting travis again :-) now it should pass travis sieb... (hopefully)

**Comment by lebrush** [comment]
Not sure why travis isn't happy this time

**Comment by OlegHahm** [comment]
Looks like a Travis-hickup - I kicked it.

**Comment by OlegHahm** [comment]
There are still some errors: https://travis-ci.org/RIOT-OS/RIOT/jobs/96284372

**Comment by lebrush** [comment]
latest changes and everything squashed

**Comment by lebrush** [comment]
Now the cortex m3 tests passed :+1:

File: boards/nucleo-f103/Makefile.features [hunk 4]
```diff
@@ -0,0 +1,8 @@
+FEATURES_PROVIDED += cpp
```
Comment (haukepetersen) [inline boards/nucleo-f103/Makefile.features]: Could you please use the same template as e.g. in `boards/stm32f4discovery/Makefile.features` for this file to keep all the Makefile.features consistent? Thx!

File: boards/nucleo-f103/include/board.h [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+ * @name Define the nominal CPU core clock in this board
+ */
+#define F_CPU               CLOCK_CORECLOCK
+
+/**
+ * @name Define the UART to be used as stdio and its baudrate
+ * @{
+ */
+#define STDIO               UART_1
+#define STDIO_BAUDRATE      (9600U)
```
Comment (haukepetersen) [inline boards/nucleo-f103/include/board.h]: Any reason for not using the default `115200` here? The boards is clocked with 72MHz, so a higher baudrate should not be an issue performance wise...

File: boards/nucleo-f103/include/board.h [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+ */
+void board_init(void);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* BOARD_H_ */
+/** @} */
+/** @} */
```
Comment (haukepetersen) [inline boards/nucleo-f103/include/board.h]: seems like one closing tag too much here

File: cpu/stm32f1/periph/spi.c [hunk 7]
```diff
@@ -210,11 +210,11 @@ int spi_transfer_byte(spi_t dev, char out, char *in)
             return -1;
     }
 
-    while (!(spi->SR & SPI_SR_TXE));
+    while (!(spi->SR & SPI_SR_TXE)) ;
     spi->DR = out;
     transferred++;
 
-    while (!(spi->SR & SPI_SR_RXNE));
+    while (!(spi->SR & SPI_SR_RXNE)) ;
```
Comment (haukepetersen) [inline cpu/stm32f1/periph/spi.c]: the introduced space here looks weird to me... Not sure if we have a convention on this?! @OlegHahm: any insights?

**Comment by haukepetersen** [comment]
port looks good, just some minor remarks/issues I can see.

File: boards/nucleo-f103/Makefile.features (hunk 4 above)
Comment (lebrush) [inline boards/nucleo-f103/Makefile.features]: ACK

File: boards/nucleo-f103/include/board.h (hunk 5 above)
Comment (lebrush) [inline boards/nucleo-f103/include/board.h]: The nucleo board includes an st-link (programmer) built in. This st-link exposes (among others) the UART of the main MCU to the host computer as a serial device. However, AFAIK on this UART forwarding only works at this baudrate. 

I never got it working in Linux neither for 9600 nor for 115200 but hooking up the signal at the pins works always fine :-)

File: cpu/stm32f1/periph/spi.c (hunk 7 above)
Comment (lebrush) [inline cpu/stm32f1/periph/spi.c]: uncrustify's cfg fault ;-)

File: boards/nucleo-f103/include/board.h (hunk 5 above)
Comment (haukepetersen) [inline boards/nucleo-f103/include/board.h]: The problem with a non-working UART in Linux on Nucleo boards is well known here, too. For some reason I am the only one who can use the UART without trouble in Linux, for most others we solved this by using external USB-to-UART adapters on the exposed UART pins... 

I would still vote for setting the default baudrate to 115k to be consistent over all Nucleo boards.

File: cpu/stm32f1/periph/spi.c (hunk 7 above)
Comment (haukepetersen) [inline cpu/stm32f1/periph/spi.c]: I see...

File: boards/nucleo-f103/include/board.h (hunk 5 above)
Comment (lebrush) [inline boards/nucleo-f103/include/board.h]: I've not a strong opinion on that... let's make it 115,2k then

Comment (haukepetersen) [inline boards/nucleo-f103/include/board.h]: cool

**Comment by haukepetersen** [comment]
once the baudrate is set to 115200: please squash. ACK once Travis is happy then

**Comment by lebrush** [comment]
@haukepetersen re-squashed

**Comment by haukepetersen** [comment]
Travis is happy -> go! Nice work!

**Comment by OlegHahm** [comment]
Congrats!

**Pull request merged**